### PR TITLE
Ofmcc 729 update user

### PR DESCRIPTION
**Pull request by jgstorey**
Summary:

- contains partial implementation for OFMCC-729 (update user, stopped due to awaiting endpoint)
- contains complete implementation for OFMCC-202 (new user, required code from 729)
- Roles are retrieved dynamically (rolesMixin.js), I think we want to change constants roles to replace ids with textual name of role.

### CODE REVIEW

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/mixins/alertMixin.js [hunk 1]
```diff
@@ -14,6 +15,11 @@ export default {
       this.addAlertNotification({ text: message, alertType: ALERT_NOTIFICATION_TYPES.SUCCESS })
     },
     setFailureAlert(message) {
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/mixins/alertMixin.js:17]: Fixed.

File: frontend/src/mixins/rolesMixin.js [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
     this.ROLES = ROLES
   },
+  computed: {
+    ...mapState(useAppStore, ['userRoles']),
+  },
+  methods: {
+    /**
+     * Get role name by id.
+     */
+    getRoleNameById(id) {
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/mixins/rolesMixin.js:18]: Fixed.

File: frontend/src/components/settings/ManageUserDialog.vue [hunk 3]
```diff
@@ -0,0 +1,245 @@
+<template>
+  <v-container>
+    <AppDialog v-model="isDisplayed" :title="addingNewUser ? 'Add new User' : 'Edit User'" :isLoading="isLoading" persistent max-width="35%" @close="closeManageUserDialog">
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/settings/ManageUserDialog.vue]: Fixed. Mock up says 'Add new user', my bad.

[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/settings/ManageUserDialog.vue]: I like that approach more... updated.

[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/settings/ManageUserDialog.vue]: Understood this for going forward, your changes addressing this merged into branch.

File: frontend/src/components/settings/ManageUserDialog.vue [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+      <template #content>
+        <v-form ref="userForm" v-model="isFormComplete" class="px-4">
+          <v-row no-gutters :class="{ 'mt-5': true, 'mb-6': !addingNewUser }">
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/settings/ManageUserDialog.vue:6]: Reason using mt and mb are adding space for when bceid input is editable or static. Without it things don't line up. Played with it awhile and came back to same place.

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**